solution/index.js

```diff
@@ -0,0 +1,310 @@
+const body= document.getElementById("body");
+let getLocal = () =>JSON.parse(window.localStorage.getItem('tasks') || 'false');  //gets local data saved
+if(!getLocal())
+{
+    getLocal= { "todo":[], "in-progress":[], "done":[] };
+}
+else
+{
+    getLocal= getLocal(); //make sure the variable is never a function and always has a object value
+}
+const saveToLocal = (data) => window.localStorage.setItem('tasks', JSON.stringify(data)); //saves new local data
+printData();
+function printData() // creates li element of existing data
+{
+    const sections= document.querySelectorAll("section");
+    for (let section of sections)
+    {
+        const ul= section.querySelector("ul");
+        if(ul.classList.contains("to-do-tasks"))
+        {
+            for(let todo of getLocal["todo"])
+            {
+                const li= document.createElement("li");
+                li.textContent = todo;
+                li.classList.add("task");
+                ul.append(li);
+            }
+        }
+        if(ul.classList.contains("in-progress-tasks"))
+        {
+            for(let inProgress of getLocal["in-progress"])
+            {
+                const li= document.createElement("li");
+                li.textContent = inProgress;
+                li.classList.add("task");
+                ul.append(li);
+            }
+        }
+        if(ul.classList.contains("done-tasks"))
+        {
+            for(let done of getLocal["done"])
+            {
+                const li= document.createElement("li");
+                li.textContent = done;
+                li.classList.add("task");
+                ul.append(li);
+            }
+         }
+    }
+}
+
+ /* event listeners */
+body.addEventListener("click",(e)=>addTask(e));
+body.addEventListener("dblclick",(e)=>editTask(e));
+body.addEventListener("mouseover", (e)=>changeSect(e));
```

Comment:
The listening event should be element specific. You solved the problem with "if(e.target.classList.contains("task"))" help but it is not an effective form